clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -461,6 +461,20 @@ class CIRGenBuilderTy : public CIRBaseBuilderTy {
     return type;
   }
 
+  mlir::cir::ArrayType getArrayType(mlir::Type eltType, unsigned size) {
+    return mlir::cir::ArrayType::get(getContext(), eltType, size);
+  }
+  
+  bool isSized(mlir::Type ty) {
+    if (ty.isIntOrFloat() ||
+        ty.isa<mlir::cir::PointerType, mlir::cir::StructType,
+                mlir::cir::ArrayType, mlir::cir::BoolType,
+                mlir::cir::IntType>())
```

Comment:
Since this is basically all types, we could probably get this from tablegen. @sitio-couto recently removed a tablegen constraint for checking that would have been helpful there. Let me get an issue for this.

Reply:
Added in #338